electron/legendary/games.ts

```diff
@@ -251,45 +273,59 @@ class LegendaryGame extends Game {
    * @public
    */
   public async addDesktopShortcut(fromMenu?: boolean) {
-    if (process.platform !== 'linux') {
+    if (process.platform === 'darwin') {
       return
     }
+
     const gameInfo = await this.getGameInfo()
-    const desktopFile = `${app.getPath('desktop')}/${gameInfo.title}.desktop`
-    const applicationsFolder = `${home}/.local/share/applications/${gameInfo.title}.desktop`
-    let shortcut;
-    const icon = await this.getIcon(gameInfo.app_name)
+    const launchWithProtocol = `heroic://launch/${gameInfo.app_name}`
+    const [ desktopFile, menuFile ] = this.shortcutFiles(gameInfo.title)
+    const { addDesktopShortcuts, addStartMenuShortcuts } = await GlobalConfig.get().getSettings()
 
     switch (process.platform) {
     case 'linux': {
-      shortcut = `[Desktop Entry]
+      const icon = await this.getIcon(gameInfo.app_name)
+      const shortcut = `[Desktop Entry]
 Name=${gameInfo.title}
-Exec=xdg-open heroic://launch/${gameInfo.app_name}
+Exec=xdg-open ${launchWithProtocol}
 Terminal=false
 Type=Application
 MimeType=x-scheme-handler/heroic;
 Icon=${icon}
 Categories=Game;
 `
-      break;
-    }
-    default:
-      logError("Shortcuts haven't been implemented in the current platform.")
-      return
+
+      if (addDesktopShortcuts || fromMenu) {
+        writeFile(desktopFile, shortcut, () => {
+          logInfo('Shortcut saved on ' + desktopFile)
+        })
+      }
+      if (addStartMenuShortcuts || fromMenu) {
+        writeFile(menuFile, shortcut, () => {
+          logInfo('Shortcut saved on ' + menuFile)
+        })
+      }
+      break
     }
-    const { addDesktopShortcuts, addStartMenuShortcuts } = await GlobalConfig.get().getSettings()
+    case 'win32': {
+      const shortcutOptions = {
+        target: launchWithProtocol,
+        icon: `${gameInfo.install.install_path}\\${gameInfo.install.executable}`,
+        iconIndex: 0
```

Comment:
the icon must be either from a DLL, the game exe file, or a .ico file, I couldn't use the same getIcon function used for linux